File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/dataset/search/ExternalDmsSearchCriteria.java
/*
 * Copyright 2016 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.dataset.search;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.AbstractObjectSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.CodeSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.SearchCriteriaToStringBuilder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.externaldms.id.IExternalDmsId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.externaldms.search.AddressSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.externaldms.search.ExternalDmsTypeSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.externaldms.search.LabelSearchCriteria;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author anttil
 */
@JsonObject("as.dto.dataset.search.ExternalDmsSearchCriteria")
public class ExternalDmsSearchCriteria extends AbstractObjectSearchCriteria<IExternalDmsId>
{

    private static final long serialVersionUID = 1L;

    public CodeSearchCriteria withCode()
    {
        return with(new CodeSearchCriteria());
    }

    public LabelSearchCriteria withLabel()
    {
        return with(new LabelSearchCriteria());
    }

    public AddressSearchCriteria withAddress()
    {
        return with(new AddressSearchCriteria());
    }

    public ExternalDmsTypeSearchCriteria withType()
    {
        return with(new ExternalDmsTypeSearchCriteria());
    }

    @Override
    protected SearchCriteriaToStringBuilder createBuilder()
    {
        SearchCriteriaToStringBuilder builder = super.createBuilder();
        builder.setName("EXTERNAL_DMS");
        return builder;
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/externaldms/search/ExternalDmsSearchCriteria.js
/**
 * @author pkupczyk
 */
define([ "require", "stjs", "as/dto/common/search/AbstractCompositeSearchCriteria", "as/dto/common/search/CodeSearchCriteria", "as/dto/externaldms/search/LabelSearchCriteria",  "as/dto/externaldms/search/AddressSearchCriteria", "as/dto/externaldms/search/ExternalDmsTypeSearchCriteria"], function(require, stjs, AbstractCompositeSearchCriteria) {
	var ExternalDmsSearchCriteria = function() {
		AbstractCompositeSearchCriteria.call(this);
	};
	stjs.extend(ExternalDmsSearchCriteria, AbstractCompositeSearchCriteria, [ AbstractCompositeSearchCriteria ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.externaldms.search.ExternalDmsSearchCriteria';
		constructor.serialVersionUID = 1;
		prototype.withCode = function() {
			var CodeSearchCriteria = require("as/dto/common/search/CodeSearchCriteria");
			return this.addCriteria(new CodeSearchCriteria());
		};
		prototype.withLabel = function() {
			var LabelSearchCriteria = require("as/dto/externaldms/search/LabelSearchCriteria");
			return this.addCriteria(new LabelSearchCriteria());
		};
		prototype.withAddress = function() {
			var AddressSearchCriteria = require("as/dto/externaldms/search/AddressSearchCriteria");
			return this.addCriteria(new AddressSearchCriteria());
		};
		prototype.withType = function() {
			var ExternalDmsTypeSearchCriteria = require("as/dto/externaldms/search/ExternalDmsTypeSearchCriteria");
			return this.addCriteria(new ExternalDmsTypeSearchCriteria());
		};
		
	}, {
		criteria : {
			name : "Collection",
			arguments : [ "ISearchCriteria" ]
		}
	});
	return ExternalDmsSearchCriteria;
})